File: development-productivity-vaadin-spring-boot/src/main/java/de/philipphauer/blog/devproductivity/ui/MyAppUI.java
package de.philipphauer.blog.devproductivity.ui;

import com.vaadin.annotations.Theme;
import com.vaadin.server.VaadinRequest;
import com.vaadin.spring.annotation.SpringUI;
import com.vaadin.ui.Grid;
import com.vaadin.ui.Label;
import com.vaadin.ui.UI;
import com.vaadin.ui.VerticalLayout;
import com.vaadin.ui.themes.ValoTheme;
import de.philipphauer.blog.devproductivity.model.Role;
import de.philipphauer.blog.devproductivity.model.User;

import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;
import java.util.stream.Stream;

@SpringUI(path = "ui")
@Theme("mytheme")
public class MyAppUI extends UI {

    private Grid<User> table = new Grid<>(User.class);
    private Label heading = new Label("Development Productivity Demo");

    @Override
    protected void init(VaadinRequest vaadinRequest) {
        getPage().setTitle("Development Productivity Demo");

        table.setSizeFull();
        table.setItems(generateDummyUsers());

        heading.addStyleName(ValoTheme.LABEL_H1);

        VerticalLayout layout = new VerticalLayout();
        layout.addComponent(heading);
        layout.addComponentsAndExpand(table);
        setContent(layout);
    }

    private List<User> generateDummyUsers() {
        Random random = new Random();
        return Stream.generate(() -> random.nextInt(9999))
                .limit(50)
                .map(uuid -> new User()
                        .setId(uuid)
                        .setFirstName("Paul")
                        .setLastName("Stark")
                        .setActive(true)
                        .setRole(Role.REGISTERED_USER)
                        .setActive(true))
                .collect(Collectors.toList());
    }
}

File: kotlin-spring-boot-vaadin-scaffolding/src/main/kotlin/de/philipphauer/blog/scaffolding/ui/MyAppUI.kt
package de.philipphauer.blog.scaffolding.ui

import com.vaadin.annotations.Theme
import com.vaadin.server.FontAwesome
import com.vaadin.server.Sizeable
import com.vaadin.server.VaadinRequest
import com.vaadin.shared.ui.label.ContentMode
import com.vaadin.spring.annotation.SpringUI
import com.vaadin.spring.navigator.SpringNavigator
import com.vaadin.ui.Label
import com.vaadin.ui.UI
import com.vaadin.ui.VerticalLayout
import com.vaadin.ui.themes.ValoTheme


@SpringUI(path = "ui")
@Theme("mytheme")
class MyAppUI(val mainContent: MainViewDisplay, navigator: SpringNavigator) : UI() {

    val navigationPresenter = NavigationPresenter(navigator)

    override fun init(request: VaadinRequest) {
        page.setTitle("Kotlin, Spring Boot, Vaadin")
        content = VerticalLayout(createHeader(), navigationPresenter.menu, mainContent).apply{
            setExpandRatio(mainContent, 1f)
            setHeight(100f, Sizeable.Unit.PERCENTAGE)
            setMargin(true)
        }
        navigationPresenter.disableCurrentMenuItem()
    }

    private fun  createHeader(): Label {
        val heading = Label("${FontAwesome.CODE.html} My Application with Kotlin, Spring Boot and Vaadin").apply {
            styleName = ValoTheme.LABEL_HUGE
            contentMode = ContentMode.HTML
        }
        return heading
    }
}